packages/block-library/src/template-part/edit.js

```diff
@@ -1,3 +1,118 @@
-export default function TemplatePartEdit() {
-	return 'Template Part Placeholder';
+/**
+ * WordPress dependencies
+ */
+import {
+	useEntityProp,
+	__experimentalUseEntitySaving,
+	EntityProvider,
+} from '@wordpress/core-data';
+import { useMemo, useCallback } from '@wordpress/element';
+import { parse } from '@wordpress/blocks';
+import { serializeBlocks } from '@wordpress/editor';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { InnerBlocks } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+const saveProps = [ 'content', 'status' ];
+function TemplatePart() {
+	const [ content, _setContent ] = useEntityProp(
+		'postType',
+		'wp_template_part',
+		'content'
+	);
+	const [ status, setStatus ] = useEntityProp(
+		'postType',
+		'wp_template_part',
+		'status'
+	);
+	const initialBlocks = useMemo( () => {
+		if ( status !== 'publish' ) {
+			// Publish if still an auto-draft.
+			setStatus( 'publish' );
+		}
+		if ( typeof content !== 'function' ) {
+			const parsedContent = parse( content );
+			return parsedContent.length ? parsedContent : undefined;
+		}
+	}, [] );
+	const [ blocks = initialBlocks, setBlocks ] = useEntityProp(
+		'postType',
+		'wp_template_part',
+		'blocks'
+	);
+	const [ isDirty, isSaving, save ] = __experimentalUseEntitySaving(
+		'postType',
+		'wp_template_part',
+		saveProps
+	);
+	const saveContent = useCallback( () => {
+		_setContent( content( { blocks } ) );
+		save();
+	}, [ content, blocks ] );
+	const setContent = useCallback( () => {
+		_setContent( ( { blocks: blocksForSerialization = [] } ) =>
+			serializeBlocks( blocksForSerialization )
```

Comment:
Is there not an equivalent selector that grants us access to the serialization of these blocks, rather than exposing the internal utility functions? Would it be possible to implement such a selector, or enhancement to an existing selector?

I'm not entirely clear what it is about the editor's serialization that we want to reuse here, vs. the raw `@wordpress/blocks` implementation, but it would be good if we can avoid exposing these internals, limit the use to either the standard public interface of a store (selectors) or the base implementation of `wp.blocks.serialize`.

Reply:
There is no selector that just serializes the inner blocks of a given block.

We could extend `getEditedPostContent`, but the name wouldn't really do it justice and the editor store doesn't really work at the blocks level.

We need this util from the editor, for the backwards compatibility reasons explained in the file:

https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/store/utils/serialize-blocks.js

I think this utility could be moved to `@wordpress/blocks` as `legacySerializeBlocks` or something. Or maybe the backwards compatibility concerns are not an issue anymore? What do you suggest?

Reply:
Perfect, we can just use `serialize` here then. I'll update it.